### PR TITLE
Clean up + autodeploy

### PR DESCRIPTION
**Pull request by Nomalah**
# Clean up + Autodeploy

Clean up the repo a little bit, add autodeploy

## Changes

- Adapted workflow to encompass deployment
- Changed files from .mts to .ts
- Changed imports from "*" to "node:*" if approprate
- Fixed style guide issues where found

## Deployment

- Created a dockerfile & container for the bot to run within

## Proof of functionality (screenshot, logs, etc)

Success in github actions workflow:
image

Bot is successfully hosted:
image

Bot successfully runs commands:
image

## Additional Notes

- Added a bunch of docs

## Checklist

- [x] Your code builds clean without any errors or warnings
- [x] Code has been formatted (clang-format, cmake-format, black, prettier, rustfmt, etc)
- [N/A] Where possible, unit tests have been added
- [x] Code is well-commented & understood

### CODE REVIEW

**Comment by DannyLeWasTaken** [comment]
LGTM

**Review by DannyLeWasTaken — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,46 +1,61 @@
...
 
-A custom-built Discord bot to run internally on our server and perform various tasks, such as:
+## About
 
--   Verify team members
--   Detect shop open/close
--   Save and back-up audit logs
--   Create countdowns for competitions
+A Discord bot used internally on our server to provide the following services:
+- Monitor and backup Audit logs (locally and on the server itself)
```
Comment (notevanrutten) [inline README.md:6]:
```suggestion
- Monitor and backup audit logs (locally and on the server itself)
```
why capital?

File: README.md [hunk 2]
```diff
@@ -1,46 +1,61 @@
...
+- Schedule events and countdown for events in specific chats
+- Monitor our shop and send/update on whether the shop is open/closed (WIP)
```
Comment (notevanrutten) [inline README.md:8]:
```suggestion
- Monitor our shop door(s) and update on whether the shop is open/closed (WIP)
```

File: README.md [hunk 3]
```diff
@@ -1,46 +1,61 @@
...
+- Semi-automatic verification of members in discord (WIP)
```
Comment (notevanrutten) [inline README.md:9]:
```suggestion
- Semi-automatic verification of members in the Discord server (WIP)
```

File: README.md [hunk 4]
```diff
@@ -1,46 +1,61 @@
...
-    DISCORD_APPLICATION_ID=""
-    DISCORD_GUILD_ID=""
-    ```
+## Development
 
-### **Build / Run Instructions:**
+### Shop Status Monitoring (WIP) - Evan
 
-1. Compile TypeScript code to JavaScript:
+We have an ESP32 at the main shop entrance monitoring if it's open or closed, if the status changes, it sends a POST request to a http server with the discord bot, which then sends a message to the Discord
```
Comment (notevanrutten) [inline README.md:39]:
```suggestion
We have an ESP32 at the main shop entrance monitoring if it's open or closed, if the status changes, it sends a POST request to an HTTP server with the Discord bot, which then sends a message to the server
```

File: README.md [hunk 5]
```diff
@@ -1,46 +1,61 @@
...
 
-    ```
-    npm run build
-    ```
+### Semi-Automatic Verification (WIP) - Danny
 
-2. Register any new `/` commands (if applicable):
+Fetch latest team roster + payment status from onedrive (TBD how), verify email, verify student number, verify student ID? If passing all requirements, the user is given the role "verified".
```
Comment (notevanrutten) [inline README.md:43]:
```suggestion
Fetch latest team roster + payment status from OneDrive (TBD how), verify email, verify student number, verify student ID? If passing all requirements, the user is given the role "Verified".
```

File: README.md [hunk 6]
```diff
@@ -1,46 +1,61 @@
...
-    npm install
-    ```
+### Build + Run
 
-2. Create `.env` file and add necessary environment variables:
+```bash
+npm install # Install dependencies
+npm run build # Transpile TypeScript to JavaScript
+node dist/deploy-commands.js # Register any new `/` commands (if applicable):
+node dist/index.mjs # Run Discord bot:
```
Comment (notevanrutten) [inline README.md:32]:
```suggestion
node dist/deploy-commands.js # Register any new slash commands (if applicable)
node dist/index.mjs # Run Discord bot
```

File: README.md [hunk 7]
```diff
@@ -1,46 +1,61 @@
...
-2. Register any new `/` commands (if applicable):
+Fetch latest team roster + payment status from onedrive (TBD how), verify email, verify student number, verify student ID? If passing all requirements, the user is given the role "verified".
 
-    ```
-    node dist/deploy-commands.mjs
-    ```
+### Countdown - Danny
 
-3. Run Discord bot:
+Initiate a countdown from a captain or lead, update the countdown every 5 minutes, and push it to the latest in the chat every day. Save countdowns locally to preserve countdowns between launches
```
Comment (notevanrutten) [inline README.md:47]:
```suggestion
Initiate a countdown from a Captain or Lead, update the countdown every 5 minutes, and push it to the latest in the chat every day. Save countdowns locally to preserve countdowns between launches.
```

File: README.md [hunk 8]
```diff
@@ -1,46 +1,61 @@
...
 
-    ```
-    node dist/index.mjs
-    ```
+### Audit Log Backup - Evan
+
+Download the audit logs every week for the past week, and upload them to discord to the audit-logs channel (hidden by default). Can also be uploaded manually via command by Captains and Leads.
```
Comment (notevanrutten) [inline README.md:51]:
```suggestion
Download the audit logs every week for the past week, and upload them to Discord to the audit-logs channel (hidden by default). Can also be uploaded manually via command by Captains and Leads.
```

File: README.md [hunk 9]
```diff
@@ -1,46 +1,61 @@
...
+
+### Deployment - Dallas
+
+This bot is run on the Embedded subsection's shop computer, it's run in a docker container and locally saved files are mounted onto the filesystem so we can use them non-volatility.
```
Comment (notevanrutten) [inline README.md:55]:
```suggestion
This bot is run on the Embedded subsection's shop computer, it's run in a docker container and locally saved files are mounted onto the filesystem to ensure non-volatility.
```

**Review by notevanrutten — CHANGES_REQUESTED** [review]
I'll assume all looks good considering I haven't worked with docker much. Just have a few changes for consistency in the readme (because OCD).

**Review by notevanrutten — APPROVED** [review]
LGTM 👍

**Pull request merged**